extension/parquet/column_writer.cpp

```diff
@@ -1137,6 +1142,37 @@ class StringStatisticsState : public ColumnWriterStatistics {
 	}
 };
 
+class StringColumnWriterState : public BasicColumnWriterState {
+public:
+	StringColumnWriterState(duckdb_parquet::format::RowGroup &row_group, idx_t col_idx)
+	    : BasicColumnWriterState(row_group, col_idx) {
+	}
+	~StringColumnWriterState() override = default;
+
+	// analysis state
+	idx_t estimated_dict_page_size = 0;
+	idx_t estimated_rle_pages_size = 0;
+	idx_t estimated_plain_size = 0;
+
+	// Dictionary
+	unordered_map<string, uint32_t> dictionary;
```

Comment:
`unordered_map<string>` will lead to many string constructions as we talked about before - should we fix it in this PR, or would you prefer to leave it for a later PR?

Reply:
Ah yes, let's do a follow-up PR then.